virttest/utils_disk.py

```diff
@@ -49,35 +49,59 @@ def copytree(src, dst, overwrite=True, ignore=''):
             shutil.copy(src_file, dst_dir)
 
 
-def is_mount(src, dst):
+def is_mount(src, dst=None, fstype=None, options=None, verbose=False,
+             session=None):
     """
     Check is src or dst mounted.
 
-    :param src: source device or directory, if None will skip to check
+    :param src: source device or directory
     :param dst: mountpoint, if None will skip to check
+    :param fstype: file system type
+    :param options: mount options should be seperated by ","
+    :param session: check within the session if given
 
-    :return: if mounted mountpoint or device, else return False
+    :return: True if mounted, else return False
     """
-    if dst and os.path.ismount(dst):
-        return dst
-    if src and (src in str(open('/proc/mounts', 'r')) or
-                src in process.system_output('losetup -a')):
-        return src
+    mount_str = "%s %s %s" % (src, dst, fstype)
+    mount_str = mount_str.replace('None', '').strip()
+    mount_list_cmd = 'cat /proc/mounts'
+
+    if session:
+        mount_result = session.cmd_output_safe(mount_list_cmd)
+    else:
+        mount_result = process.system_output(mount_list_cmd, shell=True)
+    if verbose:
+        logging.debug("/proc/mounts contents:\n%s", mount_result)
+
+    for result in mount_result.splitlines():
+        if mount_str in result:
+            if options:
+                options = options.split(",")
+                options_result = result.split()[3].split(",")
+                for op in options:
+                    if op not in options_result:
```

Comment:
How about the `nfs -> nfs4` issue?

Reply:
@luckyh Since this version I only get first 3 parameters for mount string. It would be `127.0.0.1:/home/nfs /mnt/nfs nfs` in `127.0.0.1:/home/nfs /mnt/nfs nfs4`, it will return `True` anyway. So I supposed no need to handle it as exception. Not sure for other fstype, but we can add exceptions when we meet it.